src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1394,6 +1394,29 @@ class module_sram : public module_impl
   {
       return m_scratch_pad_mem;
   }
+
+  void
+  dump_scratchpad_mem()
+  {
+    if (m_scratch_pad_mem.size() == 0) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "preemption scratchpad memory is not available");
+      return;
+    }
+
+    std::string dump_file_name = "preemption_scratchpad_mem" + std::to_string(get_id()) + ".bin";
+    std::ofstream ofs(dump_file_name, std::ios::out | std::ios::binary);
+    if (!ofs.is_open())
+      throw std::runtime_error("Failure opening file " + dump_file_name + " for writing!");
+
+    m_scratch_pad_mem.sync(XCL_BO_SYNC_BO_FROM_DEVICE);
+    auto buf = m_scratch_pad_mem.map<char*>();
+    ofs.write(buf, m_scratch_pad_mem.size());
+
+    std::stringstream ss;
+    ss << "dumped file " << dump_file_name;
+    xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", ss.str());
```

Comment:
A stream is a bit of an overkill here.
```suggestion
    std::string msg {"dumped file"};
    msg.append(dump_file_name);
    xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",msg);
```